test/support/utils_functors.h

```diff
@@ -0,0 +1,63 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _UTILS_FUNCTORS_H
+#define _UTILS_FUNCTORS_H
+
+#include <type_traits>
+#include <complex>
+
+// An arbitrary binary predicate to simulate a predicate the user providing
+// a custom predicate.
+template <typename _Tp>
+struct UserBinaryPredicate
```

Comment:
Do those functors deserve separate header?

Reply:
No, I have added these to `utils.h` and removed the new header.